routes/voice-alexa.js

```diff
@@ -0,0 +1,498 @@
+            callback(responseText, rempromptText, handlerState, false, false);
+        }  
+    });
+}
+
+function getExpectedAnswers(session) {
+	return games.check(session)
+	.then(gameIsInProgress => {
+		if(gameIsInProgress) {
+			return games.get(session).then(data => data);
+		} else {
+			return [];
+		}
+	});
+}
+
+function checkAnswer(session, answer, callback) {
+    games.answer(session, answer)
+        .then(result => {
+            spoor({
+                'category': 'GAME',
+                'action': 'answergiven',
+                'system' : {
+                    'source': 'ftlabs-correlations-game',
+                    'route': 'alexa'
+                },
+                'context' : {
+                    'product': 'ftlabs',
+                    'sessionId': session
+                }
+            });
+
+            if(result.correct === true){
+                console.log(`INFO: route=alexa; action=answergiven; sessionId=${session}; result=correct; score=${result.score};`);                
+                getQuestion(session, obj => {
+                    callback(responses.correctAnswer(result.linkingArticles[0], obj, {submitted : result.submittedAnswer, seed : result.seedPerson}), true);
+                });
+            } else {
+                console.log(`INFO: route=alexa; action=answergiven; sessionId=${session}; result=incorrect; score=${result.score}; globalHighestScore=${result.globalHighestScore}; achievedHighestScoreFirst=${result.achievedHighestScoreFirst};`);                
+                callback(responses.incorrectAnswer({expected : result.expected, seed : result.seedPerson}, result.linkingArticles[0], {score: result.score, scoreMax: result.globalHighestScore, first: result.achievedHighestScoreFirst}), false);
+            }
+        })
+    ;
+}
+
+function guessIsInAnswerList(guess, expectedAnswers) {
+    return guess.toLowerCase() === expectedAnswers[0].match ||
+    guess.toLowerCase() === expectedAnswers[1].match ||
+    guess.toLowerCase() === expectedAnswers[2].match;
+}
+
+function matchAndGetOriginalAnswer(guess, expectedAnswers) {
+    let original;
+    if (guess.toLowerCase() === expectedAnswers[0].match) {
+        original = expectedAnswers[0].original;  
+    } else if (guess.toLowerCase() === expectedAnswers[1].match) {
+        original = expectedAnswers[1].original; 
+    } else {
+        original = expectedAnswers[2].original; 
+    }
+    return original;
+}
+
+function removeSpeakTags(ssml) {
+    return ssml.replace('<speak>', '').replace('</speak>', '');
+}
+
+router.post('/', (request, response) => {
+
+    // Dummy context for Alexa handler
+    const context = {
+        fail: () => {
+            // Fail with internal server error
+            response.sendStatus(500);
+        },
+        succeed: data => {
+            response.send(data);
+        }
+    };
+
+    const alexa = Alexa.handler(request.body, context);    
```

Comment:
Does it need to be duplicated here as it's already in `handler`?

Reply:
This has been resolved